chat-widget/src/components/chatbuttonstateful/ChatButtonStateful.tsx

```diff
@@ -1,49 +1,53 @@
 import { LogLevel, TelemetryEvent } from "../../common/telemetry/TelemetryConstants";
-import React, { Dispatch, useEffect, useState } from "react";
+import React, { Dispatch, useEffect, useRef, useState } from "react";
 
 import { ChatButton } from "@microsoft/omnichannel-chat-components";
 import { Constants } from "../../common/Constants";
-import { setFocusOnElement } from "../../common/utils";
 import { ConversationState } from "../../contexts/common/ConversationState";
 import { IChatButtonControlProps } from "@microsoft/omnichannel-chat-components/lib/types/components/chatbutton/interfaces/IChatButtonControlProps";
-import { IChatButtonStatefulParams } from "./interfaces/IChatButtonStatefulParams";
 import { IChatButtonStyleProps } from "@microsoft/omnichannel-chat-components/lib/types/components/chatbutton/interfaces/IChatButtonStyleProps";
 import { ILiveChatWidgetAction } from "../../contexts/common/ILiveChatWidgetAction";
 import { ILiveChatWidgetContext } from "../../contexts/common/ILiveChatWidgetContext";
+import { ILiveChatWidgetProps } from "../livechatwidget/interfaces/ILiveChatWidgetProps";
 import { LiveChatWidgetActionType } from "../../contexts/common/LiveChatWidgetActionType";
 import { TelemetryHelper } from "../../common/telemetry/TelemetryHelper";
 import { TelemetryTimers } from "../../common/telemetry/TelemetryManager";
 import { defaultOutOfOfficeChatButtonStyleProps } from "./common/styleProps/defaultOutOfOfficeChatButtonStyleProps";
+import { setFocusOnElement } from "../../common/utils";
 import useChatContextStore from "../../hooks/useChatContextStore";
+import usePrepareStartChat from "../../hooks/usePrepareStartChat";
 
-export const ChatButtonStateful = (props: IChatButtonStatefulParams) => {
-
+export const ChatButtonStateful = (props: ILiveChatWidgetProps) => {
     const [state, dispatch]: [ILiveChatWidgetContext, Dispatch<ILiveChatWidgetAction>] = useChatContextStore();
-    const { buttonProps, outOfOfficeButtonProps, startChat } = props;
+    const { chatButtonProps, outOfOfficeChatButtonProps } = props;
     //Setting OutOfOperatingHours Flag
     const [outOfOperatingHours, setOutOfOperatingHours] = useState(state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "True");
-    
-    const outOfOfficeStyleProps: IChatButtonStyleProps = Object.assign({}, defaultOutOfOfficeChatButtonStyleProps, outOfOfficeButtonProps?.styleProps);
+    const prepareStartChat = usePrepareStartChat(props);
+
+    const ref = useRef(() => {return;});
+    ref.current = async () => {
+        TelemetryHelper.logActionEvent(LogLevel.INFO, {
+            Event: TelemetryEvent.LCWChatButtonClicked
+        });
+        
+        if (state.appStates.isMinimized) {
+            dispatch({ type: LiveChatWidgetActionType.SET_MINIMIZED, payload: false });
+        } else {
+            await prepareStartChat();
+        }
+    };
```

Comment:
I feel like we could have this functionality definition in its own file, and just import it, reducing the complexity and size of this file

Reply:
React hooks can only be called inside React components. So we'd rather add it here instead into another dummy component. 

The useRef is used here to counter the bug in FluentUI where onClick callback is not refreshed